### PR TITLE
refactor(conf_loader): separate parsing functions into parse.lua

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Keep cleaning the code of conf_loader, make it easier to maintain.

KAG-3314

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by Water-Melon — APPROVED** [review]

File: kong/conf_loader/utils.lua [hunk 1]
```diff
@@ -0,0 +1,925 @@
+local require = require
```
Comment (bungle) [inline kong/conf_loader/utils.lua:1]: Are we next going to split this utils?

Or should we do it already, aka move things in `kong.tools.xxxx`?

Comment (chronolaw) [reply kong/conf_loader/utils.lua:1]: I think that these functions are related to conf_loader only, they will not be used by other modules,
and there are already some `router/utils.lua` `clustering/utils.lua`.

**Comment by dndx** [comment]
@bungle appreciate another look at ^^

File: kong-3.6.0-0.rockspec [hunk 2]
```diff
@@ -67,6 +67,7 @@ build = {
 
     ["kong.conf_loader"] = "kong/conf_loader/init.lua",
     ["kong.conf_loader.constants"] = "kong/conf_loader/constants.lua",
+    ["kong.conf_loader.utils"] = "kong/conf_loader/utils.lua",
```
Comment (fffonion) [inline kong-3.6.0-0.rockspec:70]: i'm okay with the overall change, but the name could be improved. a module named `utils` is usually  becoming a kitchen sink. should we name it more self explantory, something like parse or parsing_utils?

Comment (chronolaw) [reply kong-3.6.0-0.rockspec:70]: rename to `parse.lua`.

**Review by bungle — APPROVED** [review]

**Pull request merged**